documentation/specs/BuildCheck/CustomBuildCheck.md

```diff
@@ -0,0 +1,142 @@
+- Inherit from the MSBuild API base class (Microsoft.Build.Experimental.BuildCheck.Check) as already done by the template as otherwise your checks won't be registered during build runtime
+- <CustomCheckName>.props file contains the intrinsic function "RegisterBuildCheck" that is picked by MSBuild and is an entry point to the check.
+- <CustomCheckName>.csproj file has a custom target `AddNuGetDlls` included for copying 3rd party assemblies in the final package
+
+### 5. Define Rule Identification
+Key components for making your check discoverable:
+- Unique Rule ID (critical for system recognition)
+- Clear, descriptive title
+- Comprehensive description
+- Actionable recommendation message
+
+### 6. Choose Build Stage for Monitoring
+Custom checks can monitor different build stages:
+- Project Evaluation Build Time (most common)
+- Access project properties
+- Track and validate configurations
+
+### 7. Implement Check Logic
+```csharp
+public sealed class MaxVersionCheck : Check
+{
+    // Define allowed versions
+    private static Dictionary<string, Version> propertiesToAllowedVersion = new Dictionary<string, Version>()
+    {
+        { "ProductVersion", new Version(6, 0, 0) }
+    };
+
+    // Unique Rule Identifier
+    private const string RuleId = "BC123";
+
+    // Define Rule with Detailed Information
+    public static CheckRule SupportedRule = new CheckRule(
+        RuleId,
+        "NoForbiddenProjectProperties",
+        "Prevent unauthorized version usage",
+        "The version '{0}' for property '{1}' is forbidden. Use version '{2}' instead.",
+        new CheckConfiguration(Severity = CheckResultSeverity.Warning));
+
+    // Registration Method
+    public override void RegisterActions(IBuildCheckRegistrationContext registrationContext)
+    {
+        registrationContext.RegisterEvaluatedPropertiesAction(EvaluatedPropertiesAction);
+    }
+
+    // Validation Logic
+    private void EvaluatedPropertiesAction(BuildCheckDataContext<EvaluatedPropertiesCheckData> context)
+    {
+        foreach (var property in propertiesToAllowedVersion)
+        {
+            if (context.Data.EvaluatedProperties.TryGetValue(property.Key, out string value))
+            {
+                if (Version.TryParse(value, out Version version) && version > property.Value)
+                {
+                    context.ReportResult(BuildCheckResult.Create(
+                        SupportedRule,
+                        ElementLocation.EmptyLocation,
+                        value,
+                        property.Key,
+                        property.Value.ToString()));
+                }
+            }
+        }
+    }
+}
+```
+
+### 8. Configure via .editorconfig
+```editorconfig
+# Custom check configuration
+build_check.BC123.severity = error
+```
+
+### 9. Package and Distribute
+- Compile as a NuGet package
+- Integrate into project build process
+- Add as a PackageReference to the checked project
+
+## Practical Considerations
```

Comment:
I think it would be nice to have a small section on how to handle errors gracefully in case the check needs to error out, and what would happen when it does (BuildCheck will emit a warning with the error message from the check, and will disable the check, so no more rules will run)